### DISCUSSION TITLE
Export macro expanded code in `schema.rs`

### DISCUSSION THREAD

**Discussion by forest1102** [diesel-rs/diesel#3256] [Ideas]
Hello, 
If there are lots of tables I need to make, every time I update or create a new table and rebuild my application, compiling takes a lot of time.  
To solve this problem, a feature to export macro expanded code in  `schema.rs` when migration runs can help.  

One reason is that expanding all `table!` macros made in schema.rs can take lots of time depending on the number of tables so using macro expanded code might help.  

Also, I know the `table!` macro makes `schema.rs` much more readable. However, I make DTO represent columns of tables, and autocomplete helps me write column names without reading schema.rs, so, I don't think the time reading `schema.rs` isn't so big.  
Then, readability in `schema.rs` is not necessary.  

Thus, compared to the necessity of the readability by using macro, the compile time is more important so expanding those macros is very appreciating.

**Comment by weiznich** [comment]
Have you some evidence that the problem is actual the expansion of those marcos? If so please add a minimal reproducible example here, so that we can either submit an upstream issue or fix the macro expansion on our side. In my experience typeck is almost always the limiting factor.

**Reply by forest1102** [reply]
Sorry, in my comparison, I found out that there isn't a big difference in compilation time between with macro and without macro.
